proc/threads.c

```diff
@@ -1649,7 +1668,15 @@ static void proc_lockUnlock(lock_t *lock)
 
 static int _proc_lockClear(lock_t *lock)
 {
-	if (lock->owner == NULL) {
+	thread_t *current = proc_current();
+
+	LIB_ASSERT(lock->owner != NULL, "lock: %s, pid: %d, tid: %d, unlock on not locked lock",
+		lock->name, (current->process != NULL) ? current->process->id : 0, current->id);
+
+	LIB_ASSERT(lock->owner == current, "lock: %s, pid: %d, tid: %d, owner: %d, unlocking someone's else lock",
+		lock->name, (current->process != NULL) ? current->process->id : 0, current->id, lock->owner->id);
+
+	if (lock->owner != current) {
```

Comment:
why was the check for lock->owner == NULL removed?

Reply:
If lock->owner == NULL then lock->owner != current (as current is never NULL). I guess we allowed to free mutex owned by a different thread, so I think I'll restore this possibility